### PR TITLE
Simplify fish baits

### LINKED ISSUE #18313: Incorrect tab for fish baits in crafting menu
Current category of fish baits is ```AMMO/OTHER```. Technically, baits _are_ ```ammo``` but it seems to be weird from player's point of view.

Any ideas about category for fish baits which fits better than current one?

### PR DESCRIPTION
**Pull request by illi-kun**
Replaces variety of fish baits by generic one as proposed in #18441.

Closes #18313 since the current PR moves fish bait recipe from `AMMO\OTHER` to `OTHER\OTHER`.

Known issues:
- All tilesets should be updated;
- It seems that migration code doesn't replace the content of active fish trap;

### CODE REVIEW

File: data/json/items/migration.json [hunk 1]
```diff
@@ -204,6 +204,26 @@
     "replace": "reloaded_shot_slug"
   },
   {
+      "id": "fish_bait_meat",
```
Comment (mugling) [inline data/json/items/migration.json]: Alphabetical ordering by `id` reduces risk of future merge conflicts

Comment (illi-kun) [inline data/json/items/migration.json]: I think you pay way too much attention to that but wev, I can update this easily.

Comment (mugling) [inline data/json/items/migration.json]: Also deterministic ordering if we implement higher-level JSON editing tools

Comment (illi-kun) [inline data/json/items/migration.json]: Fixed, ready for further review.

**Pull request merged**